src/librustc/hir/map/mod.rs

```diff
@@ -640,13 +640,13 @@ impl<'hir> Map<'hir> {
     /// Used exclusively for diagnostics, to avoid suggestion function calls.
     pub fn is_const_context(&self, hir_id: HirId) -> bool {
         let parent_id = self.get_parent_item(hir_id);
-        match self.get(parent_id) {
-            Node::Item(&Item { kind: ItemKind::Const(..), .. })
-            | Node::TraitItem(&TraitItem { kind: TraitItemKind::Const(..), .. })
-            | Node::ImplItem(&ImplItem { kind: ImplItemKind::Const(..), .. })
-            | Node::AnonConst(_)
-            | Node::Item(&Item { kind: ItemKind::Static(..), .. }) => true,
-            Node::Item(&Item { kind: ItemKind::Fn(ref sig, ..), .. }) => {
+        match self.find(parent_id) {
```

Comment:
This is weird, why could `find` ever return `None`? We should fix that if it is the case (cc @Zoxc)

Reply:
This was encountered on enum variants like `enum E { A = 1f64 }`, but I didn't dig too deep.